components/raftstore/src/store/peer.rs

```diff
@@ -4449,7 +4454,10 @@ pub trait RequestInspector {
             if apply::is_conf_change_cmd(req) {
                 return Ok(RequestPolicy::ProposeConfChange);
             }
-            if get_transfer_leader_cmd(req).is_some() {
+            if get_transfer_leader_cmd(req).is_some()
+                && !WriteBatchFlags::from_bits_truncate(req.get_header().get_flags())
+                    .contains(WriteBatchFlags::TRANSFER_LEADER_PROPOSAL)
```

Comment:
Without this change, the `TransferLeader` command after memory locks proposals triggers another round of leader transfer and results in a dead loop?

Reply:
Previously no. Because `is_valid` is set to false, so we skipped sending another lock proposal.

But now, with https://github.com/tikv/tikv/pull/11577/commits/ab86391583e6a8b362fc30ccf783ab171c4d0081#diff-7a4718b9cddd27a4ef7e563024d5283cf14b4ba3abf26edc33b974b87680a5fbL2282 change, without `msg.get_context() != TRANSFER_LEADER_COMMAND_REPLY_CTX`, there will be a dead loop.